cmake/EthCheckCXXCompilerFlag.cmake

```diff
@@ -0,0 +1,16 @@
+include(CheckCXXCompilerFlag)
```

Comment:
Can you add a comment to explain how / why is this better than the other `CheckCXXCompilerFlag`?

Reply:
@chfast can you add a comment here?

Reply:
Ah its a combination of `check_cxx_compiler_flag` and `add_compile_options`